### PR TITLE
Day 2 - Components

### PR DESCRIPTION
**Pull request by benjp009**
# 👋 Micronaut Reviewer 

## Math Magicians: setup

[See the Pull Request Objectives](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_components.md)

### Project requirements

- [x] Delete all the boilerplate from CRA (text, images, styles).
- [x] Create a directory called components.
- [x] Inside components, create a new Calculator.js file.
- [x] In Calculator.js, create a React component that matches this design:
image
- [x] Don't add any extra design.
- [x] Develop just the UI of the calculator; the math logic to make it work will be added in the next project.
- [x] Use class based components.
- [x] Once the Calculator is ready, import it and use it in your main component (App.js).
- [x] Test it in the browser; it should render without problems.

Screenshot 2022-01-03 at 15 22 08

### Pair Programming 🎳

For this week project, there is no PP required. 

Happy review :)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,26 +1,9 @@
...
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <Calculator />
```
Comment (bimbolabuari) [inline src/App.js]: As required, Let's refactor this to a class-based component 🙂

File: src/components/Calculator.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          <tr>
+            <td colSpan="2">0</td>
+            <td>.</td>
+            <td>=</td>
+          </tr>
+        </tfoot>
+      </table>
+    </div>
+  );
+}
```
Comment (bimbolabuari) [inline src/components/Calculator.js]: As required, Let's refactor this to a class-based component 👍

File: src/App.css [hunk 3]
```diff
@@ -1,39 +1,3 @@
...
-@keyframes App-logo-spin {
-  from {
-    transform: rotate(0deg);
-  }
-
-  to {
-    transform: rotate(360deg);
-  }
+body {
+  align-content: center;
```
Comment (bimbolabuari) [inline src/App.css:2]: Lets add some styling to the calculator app so it looks exactly like the required design template ✊

**Review by bimbolabuari — CHANGES_REQUESTED** [review]
Hi @benjp009, 

You have done a good job so far 🚀  

## Highlights 🎉 

- Delete all the boilerplate from CRA (text, images, styles).
- Create a directory called components.
- Inside components, create a new Calculator.js file.
-  Import calculator.js and use it in your main component (App.js).

There are some issues that you still need to work on to go to the next project but you are almost there! 🎉 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

--------


_Reviewed by  @bimbolabuari_

File: src/App.css (hunk 3 above)
Comment (benjp009) [reply src/App.css:2]: This has been done :)

File: src/App.js (hunk 1 above)
Comment (benjp009) [reply src/App.js]: It has been changed 

> class App extends React.Component {
>   render() {
>     return (
>       <Calculator />
>     );
>   }
> }

File: src/components/Calculator.js (hunk 2 above)
Comment (benjp009) [reply src/components/Calculator.js]: Same here

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @benjp009,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**